tds/routers/datasets.py

```diff
@@ -273,6 +273,8 @@ def get_dataset(id: int, rdb: Engine = Depends(request_rdb)) -> str:
     """
     Get a specific dataset by ID
     """
+    if not entry_exists(rdb.connect(), orm.Dataset, id):
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
     with Session(rdb) as session:
         result = session.query(orm.Dataset).get(id)
         return result
```

Comment:
Usually it's more efficient to just attempt to get the result and return the 404 if the result can't be fetched rather than making two queries.
Probably something like:
```
try:
    result = session.query(orm.Dataset).get(id)
    return result
catch NotFoundError:
    raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
```
Although I don't recall the exact sqlalchemy behavior/errors off the top of my head so don't just copy this directly ;).

Reply:
fixed. get returns null if no existence so doing a simple conditional